ami/ml/clustering_algorithms/cluster_detections.py

```diff
@@ -142,27 +171,50 @@ def cluster_detections(collection, params: dict, task_logger: logging.Logger = l
     logging.info(f"Using clustering algorithm: {clustering_algorithm}")
     # Creating Unknown Taxa
     update_job_progress(job, stage_key="create_unknown_taxa", status=JobState.STARTED, progress=0.0)
-    for idx, (cluster_id, cluster_detections) in enumerate(clusters.items()):
+
+    def get_cluster_name(cluster_id: int, taxon: "Taxon | None" = None, job: "Job | None" = None) -> str:
+        # if taxon and taxon.rank >= TaxonRank.ORDER:
+        #     taxon = None  # don't use in cluster name if "Lepidoptera" or higher
+
+        parts = [
+            f"Cluster {cluster_id}",
+            f"(Job {job.pk})" if job else "",
+            f"{taxon.name}?" if taxon else "",
```

Comment:
I suggest we skip making the taxon name part of the cluster name for now, I think the cluster names already look quite messy to begin with! I like that we attempt to assign a parent though (visible below the cluster name).

<img width="534" alt="Screenshot 2025-05-16 at 11 23 54" src="https://github.com/user-attachments/assets/c23795bb-b44f-4676-8dbf-0feef343f5a5" />